final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,239 @@
+        'subtitle': feed['subtitle'],
+    }
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> list:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    no_tags = re.compile('<.*?>')
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
+                break
+            articles.append(
+                SingleArticle(
+                    feed=feed['title'],
+                    title=entry['title'],
+                    date=entry['published'],
+                    link=entry['link'],
+                    summary=re.sub(no_tags, '', entry['summary']),
+                    links=[f"[{num}]: {link['href']}" for num, link in enumerate(entry['links'], 1)])
+            )
+    except KeyError as value:
+        logging.error(f"One of the entries does not have {value} key")
+        logging.info('Application ended')
+        sys.exit()
+
+    return articles
+
+
+def print_rss_json(data: list) -> None:
+    """Prints result as JSON in stdout"""
+    print(json.dumps(data, cls=EnhancedJSONEncoder))
+
+
+def print_article(data: SingleArticle) -> None:
+    """Prints a single article in stdout"""
+    print("-" * 120)
+    print(data)
+    print("-" * 120)
+
+
+def print_rss_articles(rss_articles: list) -> None:
+    """Prints source feed and articles"""
+    for article in rss_articles:
+        print_article(article)
+
+
+def check_if_verbose(args: argparse.Namespace) -> None:
+    """Checks if verbose is on and prints logs to stdout"""
+    if args.verbose:
+        with open(LOG_FILE_NAME, 'r') as fin:
+            print(fin.read(), end='')
+        logging.info('Log was printed in stdout')
+
+
+def main(args: argparse.Namespace) -> None:
+    """The main entry point of the application"""
+    rss_url = args.source
+    limit = args.limit
+
+    parsed_rss = parse_rss(rss_url)
+    logging.info('RSS URL parsed')
+
+    articles_list = get_articles(parsed_rss, limit)
+    logging.info('Articles list created')
+
+    try:
+        if args.json:
+            print_rss_json(articles_list)
+            logging.info('Articles was printed as json')
+        else:
+            print_rss_articles(articles_list)
+            logging.info(f'{limit} articles was printed')
+    except IndexError:
+        logging.error('There is no articles by this URL')
+
+    check_if_verbose(args)
+
+    logging.info('Application ended')
+
+
+def check_the_connection() -> str:
+    """Checks the connection"""
+    info = ''
+    try:
+        urllib.request.urlopen(sys.argv[1])
+    except urllib.request.HTTPError as e:
+        info = f'{e.code}: {e.reason}'
+        return info
+    except urllib.request.URLError as e:
+        info = f'{e.reason}'
+        return info
+    else:
+        return info
+
+
+def check_if_link_is_correct() -> bool:
+    """Checks if link is valid"""
+    try:
+        result = urllib.request.urlparse(sys.argv[1])
```

Comment:
If you are getting url to check using `sys.argv[1]` your script will not work if you will try to launch it like that:  
`python rss_reader.py --limit 3 "https://news.yahoo.com/rss/"`